rider/src/main/java/org/sonarlint/intellij/rider/RiderAnalysisConfigurator.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2021 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.rider
+
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.vfs.VirtualFile
+import com.jetbrains.rider.runtime.RiderDotNetActiveRuntimeHost
+import org.sonarlint.intellij.common.analysis.AnalysisConfigurator
+import org.sonarlint.intellij.common.analysis.AnalysisConfigurator.AnalysisConfiguration
+
+class RiderAnalysisConfigurator : AnalysisConfigurator {
+    override fun configure(module: Module, filesToAnalyze: Collection<VirtualFile>): AnalysisConfiguration {
+        val result = AnalysisConfiguration()
+        result.extraProperties["sonar.cs.internal.dotnetCliExeLocation"] = RiderDotNetActiveRuntimeHost.getInstance(module.project).dotNetCoreRuntime.value!!.cliExePath
```

Comment:
Not sure we want to throw an exception if `dotNetCoreRuntime.value` is null. Maybe replace `!!` with `?` ? Or don't set the property at all ?